### PR TITLE
Add docs for mesh and isosurface support

### LINKED ISSUE #3757: Document Mesh and Isosurface Support
We should update to wK docs to include a page on 3D meshes and isosurface generation. Both are great features and should be mentioned in the docs.

- [ ] Create a new page in the docs
- [ ] Create a cool GIF to show case both features
- [ ] Describe how to enable / use the features

Perhaps a short FAQ would be great:
- Which files types are support for 3D meshes?
- Are 3D meshes saved on the server? 
- Which cell are rendered using the isosurface generator? 
- What are the limitations? 
- Can I download isosurfaces as 3D meshes? 
- Can I publish and annotation with 3D meshes?

### PR DESCRIPTION
**Pull request by philippotto**
### URL of deployed dev instance (used for testing):
- https://docs.webknossos.org/v/docs-mesh/guides/tracing_ui
However, the changes do not appear there? See this PR for the actual changes.

### Issues:
- fixes #3757 

------
- [x] Updated [documentation](../blob/master/docs) if applicable
- [x] Ready for review

### CODE REVIEW

File: docs/tracing_ui.md [hunk 1]
```diff
@@ -211,12 +210,50 @@ In the `Segmentation` tab on the right-hand side, you can see the cell IDs which
...
+When **viewing a dataset** (not available in tracings) with a segmentation layer, the experimental "Render Isosurfaces" setting can be enabled within the dataset settings.
+Once the setting is enabled, webKnossos will generate isosurface meshes of the active cell and render these in the 3D viewport.
+To activate a cell, use shift+click on a segment in the tracing view.
+The isosurface is computed in chunks, which is why the isosurface is likely to appear bit by bit.
+
+![Generating isosurfaces for specific cell ids via shift+click](./images/isosurface-demo.gif)
+
+Note that the generated isosurface is not persisted for the tracing.
+Consequently, refreshing the page, will "forget" all previously generated isosurfaces.
+To persist generated isosurfaces, you can use the "download" button within the "Meshes" pane.
```
Comment (daniel-wer) [inline docs/tracing_ui.md]:
```suggestion
To persist generated isosurfaces, you can use the "Download" button within the "Meshes" pane.
```

File: docs/tracing_ui.md [hunk 2]
```diff
@@ -211,12 +210,50 @@ In the `Segmentation` tab on the right-hand side, you can see the cell IDs which
...
+That "Download" button will persist the isosurface of the active cell as an STL file to your disk.
+Using the "Import" button in the "Meshes" pane, you can re-import isosurfaces.
+
+Read more about STL imports in the following subsection.
+
+#### STL Import
+
+Apart from importing STLs which were created by webKnossos, you can also use the "Import" button within the "Meshes" tab to import arbitrary STL files.
+Such files can be created from volume data with the help of external tools, such as [Amira](https://www.fei.com/software/amira-avizo/).
+After the import, such meshes are rendered alongside the actual data in the 3D viewprt.
```
Comment (daniel-wer) [inline docs/tracing_ui.md]: I'd drop the `such` to avoid repetition

File: docs/tracing_ui.md [hunk 3]
```diff
@@ -211,12 +210,50 @@ In the `Segmentation` tab on the right-hand side, you can see the cell IDs which
...
+For each mesh, the name, position and visibility can be adapted via the mesh tab.
```
Comment (daniel-wer) [inline docs/tracing_ui.md]:
```suggestion
For each mesh, the name, position and visibility can be adapted via the "Meshes" tab.
```

File: docs/tracing_ui.md [hunk 4]
```diff
@@ -211,12 +210,50 @@ In the `Segmentation` tab on the right-hand side, you can see the cell IDs which
...
 ![A 3D Mesh visualized in webKnossos](./images/stl_mesh.png)
+![The Meshes Tab can be used to add and edit meshes.](./images/Meshes-Tab.png)
+
+#### Troubleshooting
+
+Note that the mesh and isosurface support in webKnossos is currently under heavy development.
+The functionality is currently limited by the following:
+
+- Isosurface generation only works for segmentation layers (as opposed to volume tracings). As a result, the dataset has to be opened in view mode to enable isosurfaces.
+- Isosurfaces are not persisted automatically. If you want to manually persist an isosurface, you can use the "download" button in the meshes tab.
```
Comment (daniel-wer) [inline docs/tracing_ui.md]:
```suggestion
- Isosurfaces are not persisted automatically. If you want to manually persist an isosurface, you can use the "Download" button in the meshes tab.
```

File: docs/tracing_ui.md [hunk 5]
```diff
@@ -211,12 +210,50 @@ In the `Segmentation` tab on the right-hand side, you can see the cell IDs which
...
+For each mesh, the name, position and visibility can be adapted via the mesh tab.
+In contrast to generated isosurfaces, uploaded STLs will be persisted within the tracing.
+For that reason, the import only works when opening a tracing (as opposed to just viewing a dataset).
 
 ![A 3D Mesh visualized in webKnossos](./images/stl_mesh.png)
+![The Meshes Tab can be used to add and edit meshes.](./images/Meshes-Tab.png)
+
+#### Troubleshooting
+
+Note that the mesh and isosurface support in webKnossos is currently under heavy development.
```
Comment (daniel-wer) [inline docs/tracing_ui.md]: Vielleicht
```suggestion
Note that the mesh and isosurface support in webKnossos is an ongoing development effort.
```
um das doppelte "currently" zu vermeiden.

File: docs/tracing_ui.md [hunk 6]
```diff
@@ -211,12 +210,50 @@ In the `Segmentation` tab on the right-hand side, you can see the cell IDs which
...
+#### Troubleshooting
+
+Note that the mesh and isosurface support in webKnossos is an ongoing development effort.
+The functionality is currently limited by the following:
+
+- Isosurface generation only works for segmentation layers (as opposed to volume tracings). As a result, the dataset has to be opened in view mode to enable isosurfaces.
+- Isosurfaces are not persisted automatically. If you want to manually persist an isosurface, you can use the "Download" button in the meshes tab.
+- Importing an STL which was not generated with webKnossos only works when having an annotation opened (since that mesh will be persisted).
 
+We are working on unifying the interface of meshes and isosurfaces to make the experience more intuitive.
```
Comment (hotzenklotz) [inline docs/tracing_ui.md]: :+1:

File: docs/tracing_ui.md [hunk 7]
```diff
@@ -211,12 +210,50 @@ In the `Segmentation` tab on the right-hand side, you can see the cell IDs which
...
-With the help of external tools, such as [Amira](https://www.fei.com/software/amira-avizo/), volume data can be converted to 3D meshes. These meshes can be imported into webKnossos to view them alongside the actual data.
+webKnossos offers two different ways to render segmentations as 3D meshes.
+First, meshes can simply be imported as STL files and webKnossos will render these as normal 3D objects.
+Second, webKnossos can generate isosurfaces of specific cells ad-hoc.
+These two approaches are explained in the following.
 
-To import an STL file, use the `Import STL Mesh` option in the [toolbar](#the-toolbar).
+#### Live Isosurface Generation
+
+When **viewing a dataset** (not available in tracings) with a segmentation layer, the experimental "Render Isosurfaces" setting can be enabled within the dataset settings.
```
[the remark this answers is not in the document]
Comment (hotzenklotz) [reply docs/tracing_ui.md]: No quite the right link, but the best that we hae so far: https://docs.webknossos.org/v/docs-mesh/guides/dashboard#datasets

[the remark this answers is not in the document]
Comment (hotzenklotz) [reply docs/tracing_ui.md]: https://docs.webknossos.org/v/docs-mesh/guides/tracing_ui#tracing-ui-settings

[the remark this answers is not in the document]
Comment (hotzenklotz) [reply docs/tracing_ui.md]: The setting is not yet explained in the settings section.

**Pull request merged**